www/js/survey/enketo-survey-service.js

```diff
@@ -127,6 +130,87 @@ angular.module('emission.enketo-survey.service', [
     });
   }
 
+  function _sync_getState() {
```

Comment:
software engineering: instead of copying ~ 10 functions over, move them into `ControlHelper` and invoke. That will also ensure that you don't make mistakes like introducing two `endTransititionKey`s in a row.

Reply:
@atton16 deserves a good scolding!